### ISSUE TITLE
equinox.static_field() not respected in subclass under python 3.9.0

### ISSUE THREAD

**Issue by dieterichlawson** [patrick-kidger/equinox#337]
It seems that when I have a subclass that extends a superclass with a `static_field`, the field is no longer marked as static. For example, when I run the code

```
import jax
import equinox as eqx

class A(eqx.Module):

  foo: int = eqx.static_field()

class B(A):

  def __init__(self, foo: int):
    super().__init__(foo=foo)


a = A(3)
b = B(3)
print(jax.tree_util.tree_flatten(a)[0])
print(jax.tree_util.tree_flatten(b)[0])
```

under python 3.9.0 I get the output 

```
[]
[3]
```

I don't think this happens under all versions of python 3.9, and also doesn't happen under python 3.10+. Unfortunately I am limited to python 3.9.0 in my environment. Are there any workarounds or fixes?

I'm using jax 0.4.8, jaxlib 0.4.7, and equinox 0.10.3.

**Comment by patrick-kidger** [comment]
Thanks for the report! Looks like this was a bug that snuck in with v0.10.3. Should be fixed in #338.

**Comment by patrick-kidger** [comment]
Closing as fixed in v0.10.4.
